biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -203,6 +258,84 @@ subroutine create_cohort(currentSite, patchptr, pft, nn, hite, dbh, bleaf, bfine
 
   end subroutine create_cohort
 
+  ! -------------------------------------------------------------------------------------
+
+  subroutine InitPRTCohort(new_cohort)
+
+     ! ----------------------------------------------------------------------------------
+     ! This subroutine simply allocates and attaches the correct PRT object.
+     ! The call to InitPRTVartype() performs the allocation of the variables
+     ! and boundary conditions inside the object.  It also initializes
+     ! all values as unitialized (large bogus values).
+     !
+     ! Each PARTEH allocation hypothesis has different expectations of boundary conditions.
+     ! These are specified by pointers to values in the host model. Because these
+     ! are pointers, they just need to be set once when the prt object is first initalized.
+     ! The calls below to "RegisterBCINOut", "RegisterBCIn" and "RegisterBCOut" are
+     ! setting those pointers.
+     ! -----------------------------------------------------------------------------------
+
+     !
+     ! !ARGUMENTS    
+     type(ed_cohort_type), intent(inout), target  :: new_cohort
+     type(callom_prt_vartypes), pointer :: callom_prt
+
+
+     ! Allocate the PRT class object
+     ! Each hypothesis has a different object which is an extension
+     ! of the base class.
+
+     select case(hlm_parteh_model)
+     case (prt_carbon_allom_hyp)
+        
+        allocate(callom_prt)
+        new_cohort%prt => callom_prt
+     
+     case DEFAULT
+
+        write(fates_log(),*) 'You specified an unknown PRT module'
```

Comment:
maybe add instructions on how (which file, etc.) one might have made this error...

Reply:
right, will do

Reply:
ok, will do